src/app/modules/round/round.controller.js

```diff
@@ -205,10 +206,26 @@ angular
         }
       }
 
-      // TODO Add docblock
+      /**
+       * @ngdoc method
+       * @name RoundController#cancel
+       * @kind function
+       * @methodOf app.round.controller:cancel
+       * @description
+       * Undo the last shot
+       */
       function cancel() {
-        // TODO implement
-        alert('to implement');
+        vm.isTappedNumberDisabled = false;
+        vm.number = null;
+
+        if (!vm.shots[vm.shotIndex]) {
+          vm.shotIndex --;
+          vm.shots[vm.shotIndex] = null;
```

Comment:
In entrambi i casi setti `vm.shots[vm.shotIndex] = null;`. Pensi ci sia un modo migliore di scriverlo?